spring-cloud-task-docs/src/main/asciidoc/features.adoc

```diff
@@ -0,0 +1,170 @@
+
+[[features]]
+= Features
+
+[[partintro]]
+--
+This section goes into more detail about Spring Cloud Task.  How to use it, how to
+configure it, as well as the appropriate extension points are all covered in this section.
+--
+
+[[features-lifecycle]]
+== The lifecycle of a Spring Cloud Task
+
+In most cases, the modern cloud environment is designed around the execution of processes
+that are not expected to end.  If they do, they are typically restarted.  While most
+platforms do have some method to execute a process that isn't restarted when it ends, the
+ results of that execution are typically not maintained in a consumable way.  Spring Cloud
+ Task brings the ability to execute short lived processes in an environment and record the
+ results.  This allows for a microservices architecture around short lived processes as
+ well as longer running services.
+
+While this functionality is useful in a cloud environment, the same issues can arise in a
+traditional deployment model as well.  When executing Spring Boot applications via a
+scheduler like cron, it can be useful to be able to monitor the results of the application
+after it's completion.
+
+A Spring Cloud Task the approach that a Spring Boot application can have a start and an
+end and still be successful.  Batch applications are just one example of where short lived
+processes can be helpful.  Spring Cloud Task records lifecycle events of a given task.
+
+The lifecycle consists of a single task execution.  This is a physical execution of a
+Spring Boot application configured to be a task (annotated with the `@EnableTask`
+annotation).
+
+At the beginning of a task, an entry in the `TaskRepository` is created recording the
+start event.  This event is triggered via the `ContextRefreshEvent` being triggered by
+Spring Framework.
+
+NOTE: As Spring Cloud Task is expected to consist of a single application context.  If
+multiple application contexts are used (parent/child relationships for example), the first
+`ContextRefreshEvent` that is published by Spring will be recorded as the start of the
+task.
+
+NOTE: The recording of a task will only occur upon the successful bootstrapping of an
+`ApplicationContext`.  If the context fails to bootstrap at all, the task's execution will
+not be recorded.
+
+Upon completion of all of the `*Runner`s from Spring Boot or the failure of an
```

Comment:
need to put a space between the last ` and the s for `*Runner`s.   else the second ` is ignored for *Runner